Buildings/Fluid/HeatExchangers/AirToAirHeatRecovery/BaseClasses/Effectiveness.mo

```diff
@@ -0,0 +1,165 @@
+       x1 = 0.99,
+       x2 = epsL,
+       deltaX = 1E-7);
+
+equation
+  // check if the extrapolation goes too far
+  assert(v_flow_Sup - 2*v_flow_Exh < 0 or v_flow_Exh - 2*v_flow_Sup < 0,
+    "Unbalanced air flow ratio",
+    level=AssertionLevel.warning);
+  // calculate the average volumetric air flow and flow rate ratio
+  vRat =  (v_flow_Sup + v_flow_Exh)/2/v_flow_sup_nominal;
+  // check if the extrapolation goes too far
+  assert(vRat > 0.5 and vRat < 1.3,
+    "Operatiing flow rate outside full accuracy range",
+    level=AssertionLevel.warning);
+  annotation (Icon(coordinateSystem(preserveAspectRatio=false), graphics={Text(
+          extent={{-54,28},{50,-40}},
+          textColor={28,108,200},
+          textString="eps")}), Diagram(
+        coordinateSystem(preserveAspectRatio=false)),
+    defaultComponentName="EffCal",
+Documentation(info="<html>
+<p>
+This block calculates the sensible and latent effectiveness of the heat exchanger
+for heating and cooling conditions at different air flow rates of the supply
+air stream and the exhaust air stream.
+</p>
+<p>
+It first calculates the average volumetric air flow rate through the heat exchanger by:
+</p>
+<pre>
+  v_ave = (v_sup + v_exh)/2,
+  vRat = v_ave/v_sup_nom,
+</pre>
+<p>
+where <code>v_ave</code> is the average volumetric air flow rate,
+<code>v_sup</code> is the air flow of the supply air stream,
+<code>v_exh</code> is the air flow of the exhaust air stream,
+<code>v_sup_nom</code> is the nominal air flow of the supply air stream and 
+<code>vRat</code> is the flow ratio.
+</p>
+<p>
+It then calculates the sensible and latent effectiveness by:
+</p>
+<pre>
+  epsS = y * (epsS_75 + (epsS_100 - epsS_75) * (vRat - 0.75)/0.25),
+  epsL = y * (epsL_75 + (epsL_100 - epsL_75) * (vRat - 0.75)/0.25),
+</pre>
+<p> 
+where <code>epsS</code> and <code>epsL</code> are the effectiveness
+for the sensible and latent heat transfer, respectively.
+<code>epsS_100</code> and <code>epsS_75</code> are the effectiveness 
+for the sensible heat transfer when <code>vRat</code> is 1 and 0.75, respectively.
+<code>epsL_100</code> and <code>epsL_75</code> are the effectiveness 
+for the latent heat transfer when <code>vRat</code> is 1 and 0.75, respectively.
+<code>y</code> is an effectiveness associated with the speed of a rotary wheel.
+</p>
+<p>
+<code>epsS_100</code>, <code>epsS_75</code>, <code>epsL_100</code>, and
+<code>epsL_75</code> are parameters.
+Depending on the cooling or heating mode, their values are different.
+In this model, if the supply air temperature is larger than the exhaust air
+temperature, the exchanger is considered to operate under the cooling mode;
+Otherwise, it is considered to operate under a heating mode.
+</p>
+<P>
+<b>Note:</b> The average volumetric air flow rate should be between 50% and 130% of the nominal supply air flow rate.
+In addition, the ratio of the supply air flow rate to the exhaust air flow rate should be between 0.5 and 2.
+</P>
```

Comment:
Explain why it should be that way and what happens if it is not in that range.

Moreover, as fans are typically turned down to 10% to 30% of the design flow rate, I would think this conditions is very often violated. Is there a reason we need it? How would the model be used in typical HVAC systems that have variable speed fans?

Reply:
I removed the assertion as it would trigger every time the fan switches off.